src/serialization/pkl2proto.py

```diff
@@ -0,0 +1,136 @@
+        'asciitilde'
+    ]
+
+    curveMax = 80
+    #TODO add to this list so that we know what to use
+
+    final = np.ones((len(chars), curveMax, 3,
+                     2))  # 3,2 means 3 x,y coords for quadratic bezier curve
+    final = final * 999
+
+    weird = 0
+
+    numClosedCurves = []
+    for c in range(len(chars)):
+        try:
+            # this will run per glyph
+           
+            #this is part of checking max closed curves
+            for cc in d[chars[c]]:
+                numClosedCurves.append(cc)
+                # numClosedCurves w
+
+            # curvesInGlyph = 0
+            # for counter in d[chars[c]]:
+            # curvesInGlyph +=len(d[chars[c]][counter])
+
+            # # use sum to check if there are too many curves defining a glyph
+            # if curvesInGlyph < curveMax:
+            # bezCurveIndex = 0
+            # for counter in d[chars[c]]: #this will loop through the curves/counters
+            # for curve in range(len(d[chars[c]][counter])): #loop through each curve that makes up the counter
+
+            # bez = np.asarray(d[chars[c]][counter][curve])
+            # # convert to np array to easily drop it into our data structure
+            # try:
+            # if bez.shape != (3,2):
+            # # this is just in case
+            # print("weird number of coordinates appeared:",chars[c], d[chars[c]][counter][curve], bez.shape)
+            # weird +=1
+            # final[c][bezCurveIndex] = bez
+            # bezCurveIndex+=1
+            # except:
+            # print('error with wrong num coordinates in a curve')
+
+            # # pad with 999 in the curve points as the "empty" byte
+            # # TODO perhaps think about shifting everything 999 and making the off
+            # # screen stuff at 0 so that we can reduce filesize
+
+        except:
+            pass 
+            # report the error so nohup.out will provide useful info
+            # print("Failed at reading char: ", chars[c], '\tIn font:', filename)
+            # print(sys.exc_info()) #debug message
+        
+    print( filename, ":", max(numClosedCurves))
+    # return final
+
+
+def serialize(arr, fn):
+    #take np array and turn it into bytes,
+    #store it in a protobuf
+
+    dataShape = arr.shape  #store shape of arr for reshaping
+    byt = arr.tobytes()  # pack the array
+
+    # init the protobuf
+    serializedFont = font_pb2.Font()
+
+    #shape of the file and the byte info
+    serializedFont.w = dataShape[0]
+    serializedFont.x = dataShape[1]
+    serializedFont.y = dataShape[2]
+    serializedFont.z = dataShape[3]
+    serializedFont.b = byt
+
+    fn = fn.rstrip('.p')
+    fn = fn.strip('''../../../data/fontPickles/''')
+    print(fn)
+    fn = '''../../../data/fontProto/''' + fn + '.myproto'
+    f = open(fn, 'wb')
+    f.write(serializedFont.SerializeToString())
+    f.close()
+    print('successfully saved font to:', fn)
+    return True
+
+
+def readSerializedProto(fn):
```

Comment:
You need to provide docstrings for you functions. Take a look at [what Jonny did with the ttx parser](https://github.com/ccurro/font-bakers/blob/master/src/parsettx.py#L22-L36), or [what we write for the rasterizer](https://github.com/ccurro/font-bakers/blob/master/src/rasterizer/rasterizer.py#L44-L57) for good examples.